packages/api-client-core/src/operationRunners.ts

```diff
@@ -82,36 +83,44 @@ export interface ActionRunner {
     operation: string,
     defaultSelection: FieldSelection | null,
     modelApiIdentifier: string,
+    modelSelectionField: string,
     isBulkAction: true,
     variables: VariableOptions,
     options?: SelectionOptions | null,
     namespace?: string | null
-  ): Promise<Shape extends void ? void : GadgetRecordList<Shape>>;
+  ): Promise<Shape extends void ? void : GadgetRecord<Shape>[]>;
 }
 
 export const actionRunner: ActionRunner = async <Shape = any>(
   modelManager: { connection: GadgetConnection },
   operation: string,
   defaultSelection: FieldSelection | null,
   modelApiIdentifier: string,
+  modelSelectionField: string,
   isBulkAction: boolean,
   variables: VariableOptions,
   options?: SelectionOptions | null,
   namespace?: string | null
 ) => {
-  const plan = actionOperation(operation, defaultSelection, modelApiIdentifier, variables, options, namespace);
+  const plan = actionOperation(operation, defaultSelection, modelApiIdentifier, modelSelectionField, variables, options, namespace);
   const response = await modelManager.connection.currentClient.mutation(plan.query, plan.variables).toPromise();
 
   // pass bulk responses through without any assertions since we can have a success: false response but still want
   // to process it in a similar fashion since some of the records could have been processed
   const dataPath = namespace ? [namespace, operation] : [operation];
   const mutationResult = isBulkAction ? get(response.data, dataPath) : assertMutationSuccess(response, dataPath);
 
+  // Currently, delete actions have a null selection. We do an early return for this because `hydrateRecordArray` will fail
+  // if there's nothing at `mutationResult[modelSelectionField]`, but the caller isn't expecting a return (void).
+  if (defaultSelection == null) {
```

Comment:
Not a fan of this, but it gets the job done. Tested and verified on a branch in the gadget repo.

Reply:
Makes sense! We'll probably have to evolve this over time but I dunno how exactly so send it.